symtabAPI/src/relocationEntry-elf-riscv64.C

```diff
@@ -0,0 +1,135 @@
+    switch (r) {
+        CASE_RETURN_STR(R_RISCV_NONE);
+        CASE_RETURN_STR(R_RISCV_32);
+        CASE_RETURN_STR(R_RISCV_64);
+        CASE_RETURN_STR(R_RISCV_RELATIVE);
+        CASE_RETURN_STR(R_RISCV_COPY);
+        CASE_RETURN_STR(R_RISCV_JUMP_SLOT);
+        CASE_RETURN_STR(R_RISCV_TLS_DTPMOD32);
+        CASE_RETURN_STR(R_RISCV_TLS_DTPMOD64);
+        CASE_RETURN_STR(R_RISCV_TLS_DTPREL32);
+        CASE_RETURN_STR(R_RISCV_TLS_DTPREL64);
+        CASE_RETURN_STR(R_RISCV_TLS_TPREL32);
+        CASE_RETURN_STR(R_RISCV_TLS_TPREL64);
+        CASE_RETURN_STR(R_RISCV_BRANCH);
+        CASE_RETURN_STR(R_RISCV_JAL);
+        CASE_RETURN_STR(R_RISCV_CALL);
+        CASE_RETURN_STR(R_RISCV_CALL_PLT);
+        CASE_RETURN_STR(R_RISCV_GOT_HI20);
+        CASE_RETURN_STR(R_RISCV_TLS_GOT_HI20);
+        CASE_RETURN_STR(R_RISCV_TLS_GD_HI20);
+        CASE_RETURN_STR(R_RISCV_PCREL_HI20);
+        CASE_RETURN_STR(R_RISCV_PCREL_LO12_I);
+        CASE_RETURN_STR(R_RISCV_PCREL_LO12_S);
+        CASE_RETURN_STR(R_RISCV_HI20);
+        CASE_RETURN_STR(R_RISCV_LO12_I);
+        CASE_RETURN_STR(R_RISCV_LO12_S);
+        CASE_RETURN_STR(R_RISCV_TPREL_HI20);
+        CASE_RETURN_STR(R_RISCV_TPREL_LO12_I);
+        CASE_RETURN_STR(R_RISCV_TPREL_LO12_S);
+        CASE_RETURN_STR(R_RISCV_TPREL_ADD);
+        CASE_RETURN_STR(R_RISCV_ADD8);
+        CASE_RETURN_STR(R_RISCV_ADD16);
+        CASE_RETURN_STR(R_RISCV_ADD32);
+        CASE_RETURN_STR(R_RISCV_ADD64);
+        CASE_RETURN_STR(R_RISCV_SUB8);
+        CASE_RETURN_STR(R_RISCV_SUB16);
+        CASE_RETURN_STR(R_RISCV_SUB32);
+        CASE_RETURN_STR(R_RISCV_SUB64);
+        CASE_RETURN_STR(R_RISCV_GNU_VTINHERIT);
+        CASE_RETURN_STR(R_RISCV_GNU_VTENTRY);
+        CASE_RETURN_STR(R_RISCV_ALIGN);
+        CASE_RETURN_STR(R_RISCV_RVC_BRANCH);
+        CASE_RETURN_STR(R_RISCV_RVC_JUMP);
+        CASE_RETURN_STR(R_RISCV_RVC_LUI);
+        CASE_RETURN_STR(R_RISCV_GPREL_I);
+        CASE_RETURN_STR(R_RISCV_GPREL_S);
+        CASE_RETURN_STR(R_RISCV_TPREL_I);
+        CASE_RETURN_STR(R_RISCV_TPREL_S);
+        CASE_RETURN_STR(R_RISCV_RELAX);
+        CASE_RETURN_STR(R_RISCV_SUB6);
+        CASE_RETURN_STR(R_RISCV_SET6);
+        CASE_RETURN_STR(R_RISCV_SET8);
+        CASE_RETURN_STR(R_RISCV_SET16);
+        CASE_RETURN_STR(R_RISCV_SET32);
+        CASE_RETURN_STR(R_RISCV_32_PCREL);
+        CASE_RETURN_STR(R_RISCV_IRELATIVE);
+        CASE_RETURN_STR(R_RISCV_PLT32);
+        CASE_RETURN_STR(R_RISCV_SET_ULEB128);
+        CASE_RETURN_STR(R_RISCV_SUB_ULEB128);
+        default:
+            return "Unknown relocation type";
+    }
+    return "?";
+}
+
+DYNINST_EXPORT unsigned long relocationEntry::getGlobalRelType(unsigned /*addressWidth*/, Symbol *sym) {
+    if (!sym) {
+        return R_RISCV_64;
+    }
+    if (sym->getType() == Symbol::ST_FUNCTION) {
+        return R_RISCV_JUMP_SLOT;
+    }
+    else {
+        return R_RISCV_64;
+    }
+
+	//should never return this.
+    return relocationEntry::dynrel;
```

Comment:
This will run afoul of clang's [warning](https://clang.llvm.org/docs/DiagnosticsReference.html#wunreachable-code) for unreachable code. gcc has one, as well.

Reply:
Removed.